rules/enforce-store-naming-convention/enforce-store-naming-convention.js

```diff
@@ -141,21 +160,40 @@ module.exports = {
   },
 };
 
-function reportStoreNameConventionViolation({ context, node, storeName }) {
+function reportStoreNameConventionViolation({ context, node, storeName, storeNameConvention }) {
+
+  const correctedStoreName = storeNameConvention === "prefix"
+      ? `$${storeName}`
+      : `${storeName}$`
+
   context.report({
     node,
     messageId: "invalidName",
     data: {
       storeName,
+      correctedStoreName,
+      storeNameConvention
     },
     suggest: [
       {
         messageId: "renameStore",
         data: { storeName },
         fix(fixer) {
-          return fixer.insertTextBeforeRange(node.range, "$");
+          if (storeNameConvention === "prefix") {
+            return fixer.insertTextBeforeRange(node.range, "$");
+          }
+          return fixer.insertTextAfterRange(node.range, "$");
         },
       },
     ],
   });
 }
+
+function validateStoreNameConvention(storeNameConvention) {
```

Comment:
So, this function looks like shred validation utility, which must be used in any rule with store defining.